src/ORM/Rule/ExistsIn.php

```diff
@@ -129,4 +146,22 @@ protected function _fieldsAreNull($entity, $source)
         }
         return $nulls === count($this->_fields);
     }
+
+    /**
+     * Check whether there are nullable nulls in at least one part of the foreign key.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to check.
+     * @param \Cake\ORM\Table $source The table to use schema from.
+     * @return bool
+     */
+    protected function _checkPartialSchemaNulls($entity, $source)
+    {
+        $schema = $source->schema();
+        foreach ($this->_fields as $field) {
+            if ($schema->isNullable($field) && $entity->get($field) === null) {
+                return true;
+            }
```

Comment:
> I'd still advertise to setup rules like in A) in the docs - good?

@ionas Case by case? **A)** is not the same as **B)**.
For example, **A)** would reject the insertion `INSERT INTO t2 VALUES (NULL, 9999)`, because the column `y` of `t1` only contains  `1`. `$rules->add($rules->existsIn(['y'], 'T1'));` wouldn't allow to do it.